packages/sanity/src/core/validation/validateDocumentWithReferences.ts

```diff
@@ -0,0 +1,171 @@
+import {type SanityClient} from '@sanity/client'
```

Comment:
All of this was previously handled by `document-pair/validation` 
It was moved here to make it possible to reuse by receiving `document$: Observable<SanityDocument>`

Reply:
Very nice!